frontend/src/metabase/components/Timeline.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useMemo } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import _ from "underscore";
+import styled from "styled-components";
+import moment from "moment";
+
+import { color } from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+
+const TimelineContainer = styled.div`
+  position: relative;
+  margin-left: ${props => props.leftShift}px;
+  margin-bottom: ${props => props.bottomShift}px;
+`;
+
+const TimelineItem = styled.div`
+  transform: translateX(-${props => props.leftShift}px);
+  white-space: pre-line;
+`;
+
+// shift the border down slightly so that it doesn't appear above the top-most icon
+const Border = styled.div`
+  position: absolute;
+  top: ${props => props.borderShift}px;
+  left: 0;
+  right: 0;
+  bottom: -${props => props.borderShift}px;
+  border-left: 1px solid ${color("border")};
+`;
+
+const Timeline = ({ className, items = [], renderFooter }) => {
+  const iconSize = 20;
+  const halfIconSize = iconSize / 2;
+
+  const sortedFormattedItems = useMemo(() => {
+    return items
+      .sort((a, b) => b.timestamp - a.timestamp)
+      .map(item => {
+        return {
+          ...item,
+          formattedTimestamp: moment(item.timestamp).fromNow(),
+        };
+      });
+  }, [items]);
+
+  return (
+    <TimelineContainer
+      leftShift={halfIconSize}
+      bottomShift={halfIconSize}
+      className={cx(className, "timeline")}
+    >
+      <Border borderShift={halfIconSize} />
+      {sortedFormattedItems.map((item, index) => {
+        const { icon, title, description, formattedTimestamp } = item;
+        const key = item.key == null ? index : item.key;
+
+        return (
+          <TimelineItem
+            key={key}
+            leftShift={halfIconSize}
+            className="timeline--item flex align-start justify-start mb2"
+          >
+            <Icon
+              className="timeline--item-icon text-light"
```

Comment:
I am not sure if we need BEM classes. It might be confusing since they do not add any styling and this component uses both styled-components and utility classes

Reply:
+1. If it's useful for help in testing or something like that I'm ok leaving it around, but I'd prefer to avoid the extra noise otherwise.

Reply:
I'm fine with removing them, but we should talk about it at some point so that we can try to future proof UI components for future use cases / edge cases.

When it comes to mostly-opinion-free UI components that are built to be reused in a lot of different scenarios I think offering up an escape valve for UI edge cases is super important or else you end up with a lot of subpar patterns for handling new situations that arise, whatever they may be. You'll end up with a `bodyClass` prop here, an `isSidebar` prop there, and internal classNames that get added piecemeal and thus don't conform to any sort of naming schema.

In my experience sprinkling a few BEM classes is a low effort way to aid in this, but I'm open to alternatives.